plugin-base/src/main/java/io/everytrade/server/plugin/impl/everytrade/parser/exchange/bean/OpenNodeV3.java

```diff
@@ -0,0 +1,204 @@
+    private String conversionFeesPaid;
+    private String status;
+
+    private BigDecimal baseAmount;
+    private BigDecimal quoteAmount;
+    private BigDecimal feeAmount;
+    private Currency baseCurrency;
+    private Currency quoteCurrency;
+    private Currency feeCurrency;
+    private TransactionType type;
+    Instant createdAt;
+
+    @Parsed(field = "OpenNode ID")
+    public void setOrderNodeId(String s) {
+        this.openNodeID = s;
+    }
+
+    @Parsed(field = "Date (mm/dd/yyyy UTC)")
+    public void setDate(String value) {
+        date = value;
+    }
+
+    @Parsed(field = "Time (UTC)")
+    public void setTime(String value) {
+        time = value;
+    }
+
+    @Parsed(field = "From amount")
+    public void setFromAmount(String s) {
+        fromAmount = s;
+    }
+
+    @Parsed(field = "From currency")
+    public void setFromCurrency(String s) {
+        fromCurrency = s;
+    }
+
+    @Parsed(field = "To amount")
+    public void setToAmount(String s) {
+        toAmount = s;
+    }
+
+    @Parsed(field = "To currency")
+    public void setToCurrency(String s) {
+        toCurrency = s;
+    }
+
+    @Parsed(field = "From/To exchange rate")
+    public void setFromToExchangeRate(String s) {
+        fromToExchangeRate = s;
+    }
+
+    @Parsed(field = "Conversion fees paid (BTC)")
+    public void setConversionFeesPaid(String s) {
+        conversionFeesPaid = s;
+    }
+
+    @Parsed(field = "Status")
+    public void setStatus(String s) {
+        status = s;
+    }
+
+    private Instant combineDateAndTime(String day, String time) {
+        try {
+            String stringDate = String.format("%s %s", day, time);
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MM/dd/yyyy hh-mm a");
+            final LocalDateTime localDateTime = LocalDateTime.parse(stringDate, formatter);
+            return localDateTime.toInstant(ZoneOffset.UTC);
+        } catch (Exception e) {
+            throw new DataIgnoredException("Wrong settlement date/time format. ");
+        }
+    }
+
+    @Override
+    public TransactionCluster toTransactionCluster() {
+        if (!"confirmed".equalsIgnoreCase(status)) {
+            throw new DataIgnoredException(String.format("Unfinished status: %s. ", status));
+        }
+        type = findTransactionType();
+        setUpValues();
+        return createBuySellTxs();
+    }
+
+    private TransactionType findTransactionType() {
+        if (!Currency.fromCode(toCurrency).isFiat() || Currency.fromCode(fromCurrency).isFiat()) {
+            return BUY;
+        } else {
+            return SELL;
+        }
+    }
+
+    private void setUpValues() {
+        switch (type) {
+            case BUY -> {
+                baseAmount = new BigDecimal(toAmount);
+                baseCurrency = Currency.fromCode(toCurrency);
+                quoteAmount = new BigDecimal(fromAmount);
+                quoteCurrency = Currency.fromCode(fromCurrency);
+                break;
```

Comment:
you should not need to use `break;` here . This `case BUY ->` is new swith expression syntax and doesnt need break after each case

Reply:
did you try it? IDEA itself is actually suggesting to remove it.